.codecov.yml

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# For more configuration details:
+# https://docs.codecov.io/docs/codecov-yaml
+
+# Check if this file is valid by running in bash:
+# curl -X POST --data-binary @.codecov.yml https://codecov.io/validate
+
+# Ignoring Paths
+# --------------
+# which folders/files to ignore
+ignore:
+  - "hudi-common/src/main/java/org/apache/hudi/avro/model/*"
+  - "hudi-common/src/main/java/org/apache/hudi/common/model/*"
```

Comment:
this is all the core model classes right.. why exclude them?

Reply:
they are just model classes and they only have properties + getter/setter methods. coverage does not work well with classes that have lot of member declarations since they only tag executed lines (getter/setters). Since most of the getters & setters are autogenerated or trivial code, there is no real benefit in testing them.

Reply:
@prashantwason : I agree with skipping avro-generated files for coverage. 
But even though hand-written model classes are mostly POJOs (there is some code to convert one object to another).  As these are core classes, I am inclined to keeping these on.  Let us know if you strongly think otherwise.

Reply:
No issues. I will remove the hand-written model classes.